### PR TITLE
Replace groups with environments and waves

### PR DESCRIPTION
**Pull request by LittleChimera**

### CODE REVIEW

File: api/v1alpha1/propagation_types.go [hunk 1]
```diff
@@ -63,11 +62,17 @@ func (pb PropagationBackend) TrimScheme() (string, error) {
...
+	// deployed across the deployments within the same environment. This can be used if there are multiple sites
 	// which are deployed separately but represent the same environment. In that case the rollout of a single version
 	// can be preformed across all sites before starting a rollout for a newer version.
-	GroupWith []string `json:"groupWith,omitempty"`
+	Environment string `json:"groupWith,omitempty"`
+
+	// TODO:
+	Wave int `json:"wave,omitempty"`
 
 	// Name of the deployment. This value can be used in other Propagations to determine the order in which the
```
Comment (levsha) [inline api/v1alpha1/propagation_types.go:77]: Is this description still valid?

File: api/v1alpha1/propagation_types.go [hunk 2]
```diff
@@ -140,6 +146,7 @@ type PropagationStatus struct {
 	DeploymentStatus          DeploymentStatus           `json:"deploymentStatus,omitempty"`
 	Conditions                []metav1.Condition         `json:"conditions,omitempty"`
 	DeploymentStatusesReports []DeploymentStatusesReport `json:"deploymentStatusesReports,omitempty"`
+	DeployAfter               DeployAfter                `json:"deployAfter,omitempty"`
```
Comment (levsha) [inline api/v1alpha1/propagation_types.go:149]: I don't understand how this field is filled now. Is it implemented?

**Review by levsha — APPROVED** [review]

File: api/v1alpha1/propagation_types.go (hunk 2 above)
Comment (LittleChimera) [reply api/v1alpha1/propagation_types.go:149]: it's not implemented yet. it's just an easy way to refactor this. this field should be populated once we can push the config from git repo to OCI registry.

File: api/v1alpha1/propagation_types.go (hunk 1 above)
Comment (LittleChimera) [reply api/v1alpha1/propagation_types.go]: updated description to simplify

**Pull request merged**